File: go/task-tracker/internal/task/task.go
package task

import "time"

type Task struct {
	ID          int       `json:"id"`
	Description string    `json:"description"`
	Status      string    `json:"status"`
	Priority    string    `json:"priority"` // New field: Priority (e.g., high, medium, low)
	Tags        []string  `json:"tags"`     // New field: Tags for categorization
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

// NewTask creates a new task with default values for priority and tags.
func NewTask(description, priority string, tags []string) Task {
	now := time.Now()
	return Task{
		Description: description,
		Status:      "todo",
		Priority:    priority,
		Tags:        tags,
		CreatedAt:   now,
		UpdatedAt:   now,
	}
}

File: python/task-tracker/functionality/task.py
import json
import time

class Task:
    def __init__(self, task_id, description, status="todo", created_at=None, updated_at=None):
        self.id = task_id
        self.description = description
        self.status = status
        self.created_at = created_at or time.strftime("%Y-%m-%d %H:%M:%S")
        self.updated_at = updated_at or self.created_at

    def to_dict(self):
        return {
            "id": self.id,
            "description": self.description,
            "status": self.status,
            "created_at": self.created_at,
            "updated_at": self.updated_at,
        }

    @staticmethod
    def from_dict(data):
        return Task(
            data["id"],
            data["description"],
            data["status"],
            data["created_at"],
            data["updated_at"],
        )
